### PR TITLE
Add some useful standard library functions

### PR DESCRIPTION
**Pull request by hendrikvanantwerpen**
This adds a couple of useful functions to the standard library before the next release.

### CODE REVIEW

File: src/functions.rs [hunk 1]
```diff
@@ -169,6 +173,71 @@ pub mod stdlib {
...
+                    _ => {}
+                },
+            };
+            Err(ExecutionError::FunctionFailed(
+                "eq".into(),
+                format!(
+                    "Cannot compare values of different types: {} and {}",
+                    left, right
+                ),
+            ))
```
Comment (dcreager) [inline src/functions.rs:237]: I was about to ask if you could use the derived `std::ops::Eq` impl for this, but now I can see why you didn't!

File: src/reference/functions.rs [hunk 2]
```diff
@@ -14,12 +14,19 @@
 //!
 //! # General functions
 //!
+//! ## `eq`
+//!
+//! Check if values are equal.
```
Comment (dcreager) [inline src/reference/functions.rs:19]: This deserves a sentence about how you're handling nulls.

File: src/functions.rs [hunk 3]
```diff
@@ -450,6 +519,43 @@ pub mod stdlib {
...
+                            None => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected end of format string `{}` after `{{`. Expected `{{` or `}}`.", format))),
+                        },
+                        '}' => match it.next() {
+                            Some((_, '}')) => result.push('}'),
+                            Some((i, c)) => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected character `{}` after `}}` at position {} in format string `{}`. Expected `}}`.", c, i + 1, format))),
+                            None => return Err(ExecutionError::FunctionFailed("format".into(), format!("Unexpected end of format string `{}` after `{{. Expected `}}`.", format))),
+                        },
+                        c => result.push(c),
+                    }
+                }
```
Comment (dcreager) [inline src/functions.rs:553]: If we ever want to support modifiers in the format strings, there's a [`runtime_fmt`](https://docs.rs/runtime-fmt/latest/runtime_fmt/) crate that might be useful.  No need to bring that in yet, though, this is :+1: for the simplified format strings that you're currently supporting.

File: src/functions.rs [hunk 4]
```diff
@@ -477,6 +583,24 @@ pub mod stdlib {
...
+        impl Function for Concat {
+            fn call(
+                &self,
+                _graph: &mut Graph,
+                _source: &str,
+                parameters: &mut dyn Parameters,
+            ) -> Result<Value, ExecutionError> {
+                let mut result = Vec::new();
+                while let Ok(list) = parameters.param() {
+                    result.append(&mut list.into_list()?);
```
Comment (dcreager) [inline src/functions.rs:598]: Very nice, I've always done this with `extend_from_slice`, didn't realize there was an `append` method!

**Review by dcreager — APPROVED** [review]

File: src/functions.rs (hunk 1 above)
Comment (hendrikvanantwerpen) [reply src/functions.rs:237]: I did at first, and then decided that I didn't want to be responsible for that kind of language :P

File: src/functions.rs (hunk 3 above)
Comment (hendrikvanantwerpen) [reply src/functions.rs:553]: Thanks for that reference! I was looking for something like that and couldn't find it. It was indeed the purpose to be open to richer specifiers in the future.

**Pull request merged**